components/brave_shields/browser/ad_block_regional_service_manager.cc

```diff
@@ -191,6 +192,62 @@ void AdBlockRegionalServiceManager::EnableFilterList(const std::string& uuid,
                      base::Unretained(this), uuid, enabled));
 }
 
+base::Optional<base::Value>
+AdBlockRegionalServiceManager::HostnameCosmeticResources(
+        const std::string& hostname) {
+  if (this->regional_services_.size() == 0) {
+    return base::Optional<base::Value>();
+  }
+  auto it = this->regional_services_.begin();
+  base::Optional<base::Value> first_value =
+      it->second->HostnameCosmeticResources(hostname);
+
+  for ( ; it != this->regional_services_.end(); it++) {
+    base::Optional<base::Value> next_value =
+        it->second->HostnameCosmeticResources(hostname);
+    if (first_value) {
+      if (next_value) {
+        MergeResourcesInto(&*first_value, &*next_value, false);
+      }
+    } else {
+      first_value = std::move(next_value);
+    }
+  }
+
+  return first_value;
+}
+
+base::Optional<base::Value>
+AdBlockRegionalServiceManager::HiddenClassIdSelectors(
+        const std::vector<std::string>& classes,
+        const std::vector<std::string>& ids,
+        const std::vector<std::string>& exceptions) {
+  if (this->regional_services_.size() == 0) {
+    return base::Optional<base::Value>();
+  }
+  auto it = this->regional_services_.begin();
+  base::Optional<base::Value> first_value =
+      it->second->HiddenClassIdSelectors(classes, ids, exceptions);
```

Comment:
Need to check `it`.

Reply:
Probably a better way to do it. That means we wouldn't need the `empty()` check above, right?

Reply:
Oh, I see. I didn't connect that this was safe due to the empty check above. So either way is probably fine then.